lib/mlir/Dialect/Clift/IR/ImportModel.cpp

```diff
@@ -0,0 +1,599 @@
+      return clift::PrimitiveKind::FloatKind;
+    default:
+      revng_abort();
+    }
+  }
+
+  clift::ValueType getPrimitiveType(const model::PrimitiveType &ModelType,
+                                    const bool Const) {
+    return make<clift::PrimitiveType>(getPrimitiveKind(ModelType
+                                                         .PrimitiveKind()),
+                                      ModelType.Size(),
+                                      getBool(Const));
+  }
+
+  RecursiveCoroutine<clift::TypeDefinition>
+  getTypeAttribute(const model::CABIFunctionType &ModelType) {
+    DefinitionGuard Guard(*this, ModelType.ID());
+    if (not Guard) {
+      if (EmitError)
+        EmitError() << "Recursive definition of CABIFunctionType "
+                    << ModelType.ID();
+      rc_return nullptr;
+    }
+
+    AttributeVector<clift::FunctionArgumentAttr> Args;
+    Args.reserve(ModelType.Arguments().size());
+
+    for (const model::Argument &Argument : ModelType.Arguments()) {
+      const auto Type = rc_recur getQualifiedType(Argument.Type());
+      if (not Type)
+        rc_return nullptr;
+      const llvm::StringRef Name = Argument.OriginalName();
+      const auto Attribute = make<clift::FunctionArgumentAttr>(Type, Name);
+      if (not Attribute)
+        rc_return nullptr;
+      Args.push_back(Attribute);
+    }
+
+    const auto ReturnType = rc_recur getQualifiedType(ModelType.ReturnType());
+    if (not ReturnType)
+      rc_return nullptr;
+
+    rc_return make<clift::FunctionAttr>(ModelType.ID(),
+                                        ModelType.OriginalName(),
+                                        ReturnType,
+                                        Args);
+  }
+
+  RecursiveCoroutine<clift::TypeDefinition>
+  getTypeAttribute(const model::EnumType &ModelType) {
+    DefinitionGuard Guard(*this, ModelType.ID());
+    if (not Guard) {
+      if (EmitError)
+        EmitError() << "Recursive definition of EnumType " << ModelType.ID();
+      rc_return nullptr;
+    }
+
+    auto const UnderlyingType = rc_recur getQualifiedType(ModelType
+                                                            .UnderlyingType());
+    if (not UnderlyingType)
+      rc_return nullptr;
+
+    AttributeVector<clift::EnumFieldAttr> Fields;
+    Fields.reserve(ModelType.Entries().size());
+
+    for (const model::EnumEntry &Entry : ModelType.Entries()) {
+      const auto Attribute = make<clift::EnumFieldAttr>(Entry.Value(),
+                                                        Entry.CustomName());
+      if (not Attribute)
+        rc_return nullptr;
+      Fields.push_back(Attribute);
+    }
+
+    rc_return make<clift::EnumAttr>(ModelType.ID(),
+                                    ModelType.OriginalName(),
+                                    UnderlyingType,
+                                    Fields);
+  }
+
+  RecursiveCoroutine<clift::TypeDefinition>
+  getTypeAttribute(const model::RawFunctionType &ModelType) {
```

Comment:
Adding this comment to remind me that I have to re-review this function after the latest updates to RFT that are still ongoing.